src/v/datalake/coordinator/iceberg_file_committer.cc

```diff
@@ -305,11 +307,18 @@ class table_commit_builder {
           "Adding {} files to Iceberg table {}",
           icb_files_.size(),
           table_id_);
+        // NOTE: a tag is added to the new snapshot to ensure that snapshot
+        // expiry doesn't clear the snapshot with these commit metadata
+        // properties. Tagged snapshots are by default never expired, ensuring
+        // that we retain them e.g. in case of external table updates and low
+        // snapshot retention.
```

Comment:
> Tagged snapshots are by default never expired

nitpick: does this contradict

> If no tag expiration is set, snapshot expiration will fall back on table-wide properties.

?

Reply:
Eh, technically not really since the table default is to not expire tags. But updated the comment to clarify

Reply:
dang, there's something about the way these policies fit together that I'm consistently not understanding. nothing to do with this diff though.

Reply:
If it helps, here's the table default (it matches the Java impl) https://github.com/redpanda-data/redpanda/blob/317787e74b17973f6d2a7a71681266f6b374b3ae/src/v/iceberg/remove_snapshots_action.h#L54-L55

Reply:
actually yeah, I just misinterpreted "fall back on table-wide properties" as falling back to non-tagged snapshot expiration policy rather than falling back to table-wide tag expiration policy. all good 👍